src/test/java/com/snowflake/kafka/connector/internal/IngestSdkProviderTest.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal;
+
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+import com.snowflake.kafka.connector.Utils;
+import java.util.Map;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.junit.Before;
+import org.junit.Test;
+
+public class IngestSdkProviderTest {
+  private Map<String, String> config;
+  private String connectorName;
+
+  @Before
+  // set up sunny day tests
+  public void setup() {
+    this.connectorName = "testConnector";
+
+    // config
+    this.config = TestUtils.getConfForStreaming();
+    SnowflakeSinkConnectorConfig.setDefaultValues(this.config);
+  }
+
+  @Test
+  public void testCreateClient() {
+    // setup
+    IngestSdkProvider ingestSdkProvider = new IngestSdkProvider();
+    SnowflakeStreamingIngestClient goalClient =
+        TestUtils.createStreamingClient(this.config, "KC_CLIENT_" + this.connectorName + "0");
+
+    // test
+    SnowflakeStreamingIngestClient createdClient =
+        ingestSdkProvider.createStreamingClient(this.config, this.connectorName);
+
+    // verification - very difficult (impossible?) to mock the
```

Comment:
Had trouble mocking a static factory, open to suggestions on how to test it. Ultimately I'm leaning towards changing the factory but that is out of scope for this